cmd/ls/ls_test.go

```diff
@@ -0,0 +1,92 @@
+package main
+
+import (
+	"bytes"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+)
+
+var tempDir string
+
+func TestMain(m *testing.M) {
+	setup()
+	code := m.Run()
+	teardown()
+	os.Exit(code)
+}
+
+func setup() {
+	dir, err := ioutil.TempDir("", "test")
+	if err != nil {
+		fatal(err)
+	}
+
+	tempDir = dir
+	createTempFiles(tempDir)
```

Comment:
I saw that yesterday in the Hashimoto presentation :-)

I prefer the first option. The second hack is too hacky...